src/backend/distributed/utils/datautils.c

```diff
@@ -0,0 +1,32 @@
+/*-------------------------------------------------------------------------
+ *
+ * datautils.c
+ *
+ * This file contains functions to perform useful operations on data types.
+ *
+ * Copyright (c) 2014-2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "c.h"
+
+#include "distributed/datautils.h"
+
+
+/*
+ * Allocates a new uint64 on the heap and copies the input to that space,
+ * returning a pointer to allocated memory. Useful for lists with 64-bit
+ * integers.
+ */
+uint64 *
+AllocateUint64(uint64 value)
+{
+	uint64 *allocatedValue = (uint64 *) palloc0(sizeof(uint64));
```

Comment:
> Not now, but […]

(And I understand I'm not changing anything here, just trying to understand what you mean, since it feels like I must be missing something crucial here)